scripts/build/utils/dts-process.ts

```diff
@@ -0,0 +1,67 @@
+import { writeFile } from 'node:fs/promises';
+import { dirname, join, sep } from 'node:path';
+
+import { rollup } from 'rollup';
+import { dts } from 'rollup-plugin-dts';
```

Comment:
Out of curiosity. Have you tried rolldown and `rollup-plugin-dts`'s rolldown equivalent?

Reply:
it's a stretch goal of ours to try out different bundling setups, which should be more doable now that the bundling is simpler. The paths I'm currently seeing are

1. esbuild + `rollup-plugin-dts` for types 👈 what we're migrating to now. This is how the core is built in v9.
2. tsup with default dts generation 👈 this is how the non-core packages are built in v9.
3. tsup with [experimental dts generation](https://tsup.egoist.dev/#generate-declaration-file)
4. tsdown

Reply:
We have not. We're not doing a spike for new tools as part this project.